File: src/MetaDSL/test/models/dotnet/Role.cs
using System;
using System.ComponentModel.DataAnnotations;

public class Role
{
    [Key]
    public Guid Id { get; private set; }  // L'ID est privé, accès en lecture seule

    public string Name { get; private set; }  // Le nom du rôle est en lecture seule pour tout le monde

    public Role(string name)
    {
        Id = Guid.NewGuid();
        Name = name;
    }
}

File: src/MetaDSL/test/models/ts/Role.ts
export class Role {
    public readonly id: string;  // L'ID est en lecture seule

    public readonly name: string;  // Le nom du rôle est en lecture seule pour tout le monde

    constructor(id: string, name: string) {
        this.id = id;
        this.name = name;
    }
}
